lib/fosdick/order.rb

```diff
@@ -198,7 +198,12 @@ def handle_error(response)
 
       case errors[0]
       when "Invalid"
-        raise InvalidError, errors.last
+        last_error = errors.last
+        if last_error.include?('Duplicate ExternalId')
+          raise DuplicateError, last_error
```

Comment:
It'd be amazing if we could extract the duplicate order ID and make it a property of the error. Something like:

```ruby
duplicate_id = last_error.scan(/^Duplicate ExternalID: (.+)$/)[0][0]
raise DuplicateError, duplicate_id
```

```ruby
class DuplicateError < ApiError
  attr_reader :external_id

  def initialize(external_id)
    @external_id = external_id
  end

  def message
    "Duplicate Fosdick order: #{external_id}"
  end
end
```

I realize that it will already show in the `message`, but wanted to show an alternative way to put properties on an error object.

Reply:
Ahh, that's a nice little example - doesn't seem super useful in this instance though